src/main/java/net/mcreator/workspace/elements/VariableElement.java

```diff
@@ -38,11 +43,28 @@
 	}
 
 	public VariableElementType getType() {
-		return type;
+		return getVariableFromType(type);
+	}
+
+	public static VariableElementType getVariableFromType(String type) {
+		for (VariableElementType varType : VARIABLE_CACHE.keySet()) {
+			if(varType.getBlocklyVariableType().equalsIgnoreCase(type) || varType.getName().equalsIgnoreCase(type)) {
+				return varType;
+			}
+		}
+		return null;
+	}
+
+	public static Set<VariableElementType> getVariables() {
+		return VARIABLE_CACHE.keySet();
+	}
+
+	public static void addVariableTypeToCache(VariableElementType var) {

```

Comment:
I see what you mean here. I think you could move this to variableloader and call this VARIABLE_LIST, not VARIABLE_CACHE